tests/bsim/bluetooth/mesh/src/test_beacon.c

```diff
@@ -1370,6 +1370,305 @@ static void test_rx_priv_interleave(void)
+
+	random_field = net_buf_simple_pull_le64(buf);
+	ASSERT_EQUAL(hash, pp_hash_calc(test_net_key, random_field, (uint16_t *)&tx_cfg.addr));
+
+	k_sem_give(&observer_sem);
+}
+
+static void priv_multi_net_id_scan_cb(const bt_addr_le_t *addr, int8_t rssi, uint8_t adv_type,
+				      struct net_buf_simple *buf)
+{
+	static bool net1_rcv, net2_rcv;
+
+	if (!pp_type_check(BEACON_TYPE_PRIVATE_NET_ID, adv_type, buf)) {
+		/* Wrong message type */
+		return;
+	}
+
+	uint64_t hash = net_buf_simple_pull_le64(buf);
+
+	random_field = net_buf_simple_pull_le64(buf);
+
+	if (hash == pp_hash_calc(test_net_key, random_field, NULL)) {
+		net1_rcv = true;
+	} else if (hash == pp_hash_calc(test_net_key_secondary, random_field, NULL)) {
+		net2_rcv = true;
+	}
+
+	/* Both networks has been received*/
+	if (net1_rcv && net2_rcv) {
+		k_sem_give(&observer_sem);
+	}
+}
+
+static bool wait_for_priv_beacon(bt_le_scan_cb_t cb, uint16_t wait_ms)
+{
+	struct bt_le_scan_param scan_param = {
+			.type       = BT_HCI_LE_SCAN_PASSIVE,
+			.options    = BT_LE_SCAN_OPT_NONE,
+			.interval   = BT_MESH_ADV_SCAN_UNIT(1000),
+			.window     = BT_MESH_ADV_SCAN_UNIT(1000)
+	};
+	bool received = false;
+	int err;
+
+	err = bt_le_scan_start(&scan_param, cb);
+	if (err && err != -EALREADY) {
+		FAIL("starting scan failed (err %d)", err);
+	}
+
+	err = k_sem_take(&observer_sem, K_MSEC(wait_ms));
+	if (!err) {
+		received = true;
+	} else {
+		LOG_WRN("Didn't receive beacon in time (err: %d)", err);
+	}
+
+	err = bt_le_scan_stop();
+	if (err && err != -EALREADY) {
+		FAIL("stopping scan failed (err %d)", err);
+	}
+
+	return received;
+}
+
+static void rx_priv_common_init(uint16_t wait)
+{
+	int err;
+
+	bt_mesh_test_cfg_set(&rx_cfg, wait);
+	bt_mesh_crypto_init();
+	k_sem_init(&observer_sem, 0, 1);
+
+	err = bt_enable(NULL);
+	if (err) {
+		FAIL("Bluetooth init failed (err %d)", err);
+		return;
+	}
+}
+
+static void tx_priv_common_init(uint16_t wait)
+{
+	int err;
+
+	bt_mesh_test_cfg_set(NULL, wait);
+	bt_mesh_device_setup(&prov, &prb_comp);
+	provision(&tx_cfg);
+
+	/* Disable GATT proxy */
+	err = bt_mesh_gatt_proxy_set(BT_MESH_GATT_PROXY_DISABLED);
+	if (err) {
+		FAIL("Failed to disable gatt proxy (err %d)", err);
+	}
+}
+
+static void test_tx_priv_net_id(void)
+{
+	int err;
+
+	tx_priv_common_init(610);
+
+	/* Enable private GATT proxy */
+	err = bt_mesh_priv_gatt_proxy_set(BT_MESH_GATT_PROXY_ENABLED);
+	if (err) {
+		FAIL("Failed to set private gatt proxy (err %d)", err);
+	}
+
+	PASS();
+}
+
+static void test_rx_priv_net_id(void)
+{
+	uint64_t old_random;
+
+	rx_priv_common_init(610);
+
+	/* Scan for first net ID */
+	ASSERT_TRUE(wait_for_priv_beacon(priv_net_id_scan_cb, 1000));
+	old_random = random_field;
+
+	/* Wait for 10 minutes, then scan for second net
+	 * ID and verify that random field has changed
+	 */
+	k_sleep(K_SECONDS(600));
+	ASSERT_TRUE(wait_for_priv_beacon(priv_net_id_scan_cb, 1000));
+	if (old_random == random_field) {
```

Comment:
> 	/* Wait for 10 minutes, then scan for second net
	 * ID and verify that random field has changed
	 */

Different random doesn't guarantee that it was issued by another network. Or the comment is incorrect?

Reply:
I guess it could say "then scan for another net ID beacon".